### DISCUSSION TITLE
端口跳跃支持

### DISCUSSION THREAD

**Discussion by LLIce** [kkkgo/PaoPaoGateWay#86] [问和答] [answered]
dns跟gw用着都很舒服，看到meta内核也支持Hysteria 2，折腾了下搭了个节点也能正常使用。
但会碰到封端口的问题，然后去官方文档看到有端口跳跃这功能，用桌面客户端配置也能正常使用，但是配置到gw就卡在了解析custom这里
![0$V_VASN0UM13WMZ@HJB)OO](https://github.com/kkkgo/PaoPaoGateWay/assets/13741076/790adba5-b7ed-48df-9064-19d02e9745e8)
然后我翻了下meta那边
![image](https://github.com/kkkgo/PaoPaoGateWay/assets/13741076/1ef83215-73d5-490f-b187-f67f9ffa970e)
确实是已经支持端口跳跃了。
想问下，这个能解决吗。

**Comment by LLIce** [comment]
proxies:
  - name: 'hy2'
    type: hysteria2
    server: example.com
    port: “443,20000-50000” #解析到这里卡住了
    password: 
    transport:
      udp:
        hopInterval: 30s


![image](https://github.com/kkkgo/PaoPaoGateWay/assets/13741076/c864982b-2fc8-45b8-a576-7416058ee78f)

**Comment by kkkgo** [comment]
你可以用docker定制核心。

**Reply by kkkgo** [reply, marked as answer]
至于解析配置和协议支持，跟ppgw无关，你也可以去下载二进制的核心，手动用命令行测试配置文件是否能解析正确。

**Reply by LLIce** [reply]
之前还以为ppgw有解析，研究了下配置文件发现在端口后面加双引号带范围，虚拟机开机就能正确解析了。谢谢作者。
  - {name: hy2, server: example.com, port: 443,"21000-50000", type: hysteria2,  password: ps, transport:udp:hopInterval: 30s}
